lib/pengine/common.c

```diff
@@ -283,14 +283,14 @@ static pcmk__cluster_option_t pe_opts[] = {
 };
 
 void
-pe_metadata(void)
+pe_metadata(pcmk__output_t *out)
 {
     char *s = pcmk__format_option_metadata("pacemaker-schedulerd",
                                            "Pacemaker scheduler options",
                                            "Cluster options used by Pacemaker's scheduler"
                                                " (formerly called pengine)",
                                            pe_opts, PCMK__NELEM(pe_opts));
-    printf("%s", s);
+    out->output_xml(out, "metadata", s);
```

Comment:
I think it's so cool we can have this one line, and automatically the binary supports different text vs xml output with no additional coding. The output model will pay dividends a long time, I think

Reply:
Incidentally, is "metadata" the best tag to use here?  Maybe "option-metadata" might be better?  Right now, the resulting XML looks like:
```
<pacemaker-result api-version="2.14" request="/home/clumens/src/pacemaker/daemons/schedulerd/.libs/pacemaker-schedulerd --output-as xml metadata">
  <metadata><![CDATA[<?xml version="1.0"?><!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
<resource-agent name="pacemaker-schedulerd">
  <version>1.1</version>
```
Note the tag just before the CDATA block begins.

Reply:
metadata is fine, it doesn't really matter. But that reminds me we need to bump PCMK__API_VERSION and update the schema

Reply:
I couldn't find anywhere that we were using a schema for the daemons, so I'd avoided doing anything along those lines.

Reply:
For this purpose, the daemon is just another command-line tool. It would just be a schema for <metadata>

Reply:
Alternatively, you could make the xml output follow the schema for crm_resource --show-metadata

Reply:
metadata is just <text /> everywhere, because I couldn't find any good way to have a schema for XML embedded in XML.

Reply:
All the daemons will have the same format output once converted to the output model, so I wouldn't make the schema schedulerd-specific. I now see crm_resource-2.11.rng already has a schema for \<metadata\>, so actually I was wrong -- we didn't need to do anything with the schemas, it should already validate.

Reply:
Yeah, though it feels like it's validating by accident.  I'm not sure I'll ever remember that the reason a daemon's output validates is because of something going on over in crm_resource.rng.  I don't have any other suggestions than to make a single daemon schema.

Reply:
There's nothing crm_resource-dependent about the crm_resource RNGs, it's just a convenient tag to point to a tool that uses it. It's just a schema for XML that can be returned by the Pacemaker APIs, whether tool, daemon, or libpacemaker. Reusing schemas makes sense when it's the same logical content.